src/Stratis.Features.FederatedPeg/Collateral/SmartContractCollateralPoARuleRegistration.cs

```diff
@@ -23,17 +23,20 @@ public class SmartContractCollateralPoARuleRegistration : SmartContractPoARuleRe
 
         private readonly ICollateralChecker collateralChecker;
 
+        private readonly CheckCollateralFullValidationRule checkCollateralFullValidationRule;
+
         private readonly IDateTimeProvider dateTime;
 
         public SmartContractCollateralPoARuleRegistration(Network network, IStateRepositoryRoot stateRepositoryRoot, IContractExecutorFactory executorFactory,
             ICallDataSerializer callDataSerializer, ISenderRetriever senderRetriever, IReceiptRepository receiptRepository, ICoinView coinView,
             IEnumerable<IContractTransactionPartialValidationRule> partialTxValidationRules, IEnumerable<IContractTransactionFullValidationRule> fullTxValidationRules,
-            IInitialBlockDownloadState ibdState, ISlotsManager slotsManager, ICollateralChecker collateralChecker, IDateTimeProvider dateTime)
+            IInitialBlockDownloadState ibdState, ISlotsManager slotsManager, ICollateralChecker collateralChecker, IDateTimeProvider dateTime, CheckCollateralFullValidationRule checkCollateralFullValidationRule)
```

Comment:
Can't we rather inject `CounterchainNetworWorkwrapper` here as opposed to `CheckCollateralFullValidationRule` seeing as we already have it in the DI? Then is saves us from adding to the DI.

Then construct the rule as per usual but with the counter chain network.

Reply:
Doing it this way requires smart contracts to know less about constructing the collateral rule, which I think is a good thing.